implementations/rust/ockam/ockam_entity/src/credential/workers/listener.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{
+    get_secure_channel_participant_id, CredentialProtocolMessage, CredentialSchema, IssuerWorker,
+    Profile, SecureChannelTrustInfo, TrustPolicy, TrustPolicyImpl,
+};
+use async_trait::async_trait;
+use ockam_core::{Address, Result, Routed, Worker};
+use ockam_node::Context;
+
+pub struct ListenerWorker {
```

Comment:
`CredentialProtocolListener`? Name is too generic

Reply:
I think we should consider making this type private, the name is then not a concern anymore, right?